x-pack/plugins/actions/README.md

```diff
@@ -529,10 +537,10 @@ The ServiceNow action uses the [V2 Table API](https://developer.servicenow.com/a
+| Property              | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   | Type                |
+| --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------- |
+| apiUrl                | ServiceNow instance URL.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      | string              |
+| incidentConfiguration | Case configuration object. The object should contain an attribute called `mapping`. A `mapping` is an array of objects. Each mapping object should be of the form `{ source: string, target: string, actionType: string }`. `source` is the Case field. `target` is the ServiceNow field where `source` will be mapped to. `actionType` can be one of `nothing`, `overwrite` or `append`. For example the `{ source: 'title', target: 'short_description', actionType: 'overwrite' }` record, inside mapping array, means that the title of a case will be mapped to the short description of an incident in ServiceNow and will be overwrite on each update. | object _(optional)_ |
```

Comment:
Seems like you are renaming an existing config value of `casesConfiguration` to `incidentConfiguration`.  If you do that, any existing actions with `casesConfiguration` will be broken - they won't validate anymore.

I don't believe we had to deal with this yet, but I think the way to do it would be like the way we treat kibana.yml config values that we "rename".  Basically, you need to deprecate the old one, and not delete it.  In code that checks the value, you will potentially have to check both (use the new one unless it's not set,  otherwise use the old one).  @mikecote other thought?

I'm not sure if we would support some way for these to be migrated.

Reply:
Good catch @pmuellr, I saw the same rename got done [in ServiceNow too](https://github.com/elastic/kibana/pull/68464). From what I see, the impact is that the connector would "lose" the field mappings because `incidentConfiguration` would be empty for existing ServiceNow connectors (possible bug in Cases app?) and the user would have to re-configure the mappings. I'm not sure how it would validate when updating 🤔 

One way I see this being fixed since we're not GA yet is with a saved object migration in the actions plugin that moves the value from `casesConfiguration` to `incidentConfiguration` (including decrypt / encrypt process). In a post GA world, @pmuellr is correct and we'd probably have to think of a non-breaking way to support both until the next major where possible.

I'm indifferent what approach is done (migrate or revert to `casesConfiguration` or make users re-configure mappings). Since there's a phase 2 to remove this value from connectors, we could always skip it until the field gets removed.

Reply:
@pmuellr @mikecote I followed [ServiceNow PR](https://github.com/elastic/kibana/pull/68464) example. As mapping for the moment is fixed to a default one, creating or updating a case or validating the mapping will not be a problem. What Cases will loose is the appended text (Example: `A title in Jira (created at 2020-04-22T18:18:03.896Z by Christos Nasikas)`) in titles and descriptions. I totally agree with what you are saying but I think it's ok if some users lose temporarily that functionality as in phase 2 we gonna append this information beforehand in Cases. @XavierM  What are your thoughts on that?

Reply:
As of today, this mapping is static meaning nobody can change it. So we can always default back but I think it will be better to do a migration. @cnasikas I can show you how to do it if you do not know, but this one seems pretty simple.